### PR TITLE
Update gomobile

### PR DESCRIPTION
**Pull request by chrisnojima**
While our internet was down/slow i started to work to make the go build faster
It does the following

- [x] Update gomobile. We had a pretty old version of this vendored and this updates it. 
  - Code wise on our side they changed the namespace (java go.keybase.blah is now just keybase.blah) (and on ios its KeybaseBlah vs GoKeybaseBlah).
  - Return values actually are the return values in objc vs being a pointer in the arguments (s = KeybaseBlah(&err) vs GoKeybaseBlah(&s, &err))
  - Requires ndk on android now
- [x] Update the rn-gobuild-* scripts to try and build w/o doing a gomobile first. gomobile is very slow and often we don't need it. Also set an env var which is really required for it to notice its still up to date
- [x] Shallow clone the repos (kbfs/client) so if you don't use the local its faster
- [x] Killed keyListener on ios since we don't use this

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits cf270416ee26...8f9c870a5eee introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Profile-Your_Profile_Broken_Proof_Menu** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1505936878252-cf270416ee26-8f9c870a5eee-linux/Profile-Your_Profile_Broken_Proof_Menu-cf270416ee26.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1505936878252-cf270416ee26-8f9c870a5eee-linux/Profile-Your_Profile_Broken_Proof_Menu-8f9c870a5eee.png)  
   Profile-Your_Profile_Broken_Proof_Menu comparison
</summary>

File: go/vendor/vendor.json [hunk 1]
```diff
@@ -665,10 +665,10 @@
 			"revisionTime": "2016-12-08T01:05:43Z"
 		},
 		{
-			"checksumSHA1": "/7kE7RA30R1A9nZ3mXbbfA/UDQY=",
+			"checksumSHA1": "yZW/GZU87G6S/SaV92usFTFwSds=",
 			"path": "golang.org/x/mobile",
-			"revision": "e99a906c3a3ac5959fa4b8d08f90dd5f75d3b27c",
-			"revisionTime": "2016-09-07T14:02:52Z",
```
Comment (chrisnojima) [inline go/vendor/vendor.json]: almost a year old!

File: shared/react-native/gobuild.sh [hunk 2]
```diff
@@ -25,6 +25,9 @@ kbfs_dir="$GOPATH0/src/github.com/keybase/kbfs"
 GOPATH="$tmp_gopath"
 echo "Using temp GOPATH: $GOPATH"
 
+# if we don't set this gomobile init get confused
+GOMOBILE="$GOPATH/pkg/gomobile"
```
Comment (chrisnojima) [inline shared/react-native/gobuild.sh]: if this is missing the go build will always assume its out of date!

File: shared/react-native/gobuild.sh [hunk 3]
```diff
@@ -36,7 +39,7 @@ go_kbfs_dir="$tmp_gopath/src/github.com/keybase/kbfs"
 
 if [ ! "$local_client" = "1" ]; then
   echo "Getting client (via git clone)... To use local copy, set LOCAL_CLIENT=1"
-  (cd "$GOPATH/src/github.com/keybase" && git clone https://github.com/keybase/client)
+  (cd "$GOPATH/src/github.com/keybase" && git clone --depth=1 https://github.com/keybase/client)
```
Comment (chrisnojima) [inline shared/react-native/gobuild.sh]: don't clone the whole history!

File: shared/react-native/gobuild.sh [hunk 4]
```diff
@@ -97,14 +89,44 @@ keybase_build=${KEYBASE_BUILD:-$build}
...
+{
+  echo "Build gomobile..."
+  (cd "$gomobile_path" && go build -o "$GOPATH/bin/gomobile")
+  # The gomobile binary only looks for packages in the GOPATH,
+  rsync -pr --ignore-times "$vendor_path/" "$GOPATH/src/"
+  echo "Doing gomobile init"
+  if [ "$arg" = "ios" ]; then
+    "$GOPATH/bin/gomobile" init
+  elif [ "$arg" = "android" ]; then
+    "$GOPATH/bin/gomobile" init -ndk $ANDROID_HOME/ndk-bundle
```
Comment (chrisnojima) [inline shared/react-native/gobuild.sh]: ndk required now

**Comment by chrisnojima** [comment]
@keybase/react-hackers this should speed up our go build
please do a `yarn rn-gobuild-ios and rn-gobuild-android` and build the apps
once this merges we should kick off a bot build also

**Pull request merged**